OEF-Social_Service/OEF-Social_Service.DataAccess/Data/Services/FollowService.cs

```diff
@@ -3,36 +3,164 @@
+
+            using (_session)
+            {
+                var query = await _session.RunAsync(statementText.ToString(), statementParameters);
+                Console.Write(query);
+                
+            }
+        }
+
+        public async Task SendRequest(string person1, string person2)
+        {
+            if(person1 == person2)
+            {
+                return;
+            }
+            else
+            {
+                var test = DoesRelationExist(person1, person2).Result;
+                if (test == false)
+                {
+                    var statementText = new StringBuilder();
+                    statementText.Append("MATCH (p1:Person), (p2:Person) WHERE p1.Firstname = $firstname AND p2.Firstname = $firstname2 CREATE (p1)-[p:Request_Send] ->(p2)");
+                    var statementParameters = new Dictionary<string, object>
+            {
+                {"firstname", person1}, 
+                {"firstname2", person2}
+            };
+                    using (_session)
+                    {
+                        //_session.LastBookmark.Values(null);
+                        var i = await _session.RunAsync(statementText.ToString(), statementParameters);
+                    }
+                }
+                else
+                {
+                    return;
+                }
+
+            }
+
+
+
         }
 
-        public void printGreeting(string message)
+        public async Task<string> GetRequests(string person)
         {
+            var data = new List<Person>();
+            var statementText = new StringBuilder();
+            statementText.Append("Match (user:Person)-[r:Request_Send]->({Firstname: $firstname}) Return user");
+            var statementParameters = new Dictionary<string, object>
+            {
+                {"firstname", person},
+            };
+            
+            using (_session)    
+            {
+                var query = await _session.RunAsync(statementText.ToString(), statementParameters);
+                var result = await query.ToListAsync();
+                var i = JsonSerializer.Serialize(result);
+                Console.WriteLine(i);
+
+                return i;
+            }
+        }
+        public async Task<bool> DoesRelationExist(string person1, string person2)
+        {
+            var statementText = new StringBuilder();
+            statementText.Append("Match (p:Person {Firstname:$firstname}), (b:Person {Firstname: $firstname2}) RETURN EXISTS((p)-[]->(b))");
+            var statementParameters = new Dictionary<string, object>
+            {
+                {"firstname", person1},
+                {"firstname2", person2}
+            };
+
+            using (replicaSession)
+            {
+                var query = await replicaSession.RunAsync(statementText.ToString(), statementParameters);
+                var result = query.ToListAsync();
+                foreach (var item in result.Result[0].Values)
```

Comment:
kun je result.Result niet veranderen in een andere datastructuur als je toch enkel de eerste key verwacht? (Result[0] naar Result)

Reply:
Besloten om het niet te doen. Maar inderdaad is dit wel een goeie puntje om later mee te nemen.